### PR TITLE
Adds a basic filter feature

### PR DESCRIPTION
**Pull request by solomon-b**
This PR adds a basic filter feature for viewing your PDF collection. All `Paper` fields are squashed together into a single `Text` value per `Paper` and then a fuzzy finder is used for filtering.

It also does some refactoring on the overall `lib` file:
- Moved various functions around to group them more semantically
- Simplified the `FriendState`/`FriendView` functions.
- Resolved all warnings

### CODE REVIEW

**Comment by chessai** [comment]
This LGTM, assuming it works. 

A stylistic nitpick: instead of `flip (,)`, you can use `-XTupleSections`, or a lambda. Up to you.

**Comment by solomon-b** [comment]
Good call on the Tuple Section. I never think to use them but here it makes perfect sense.

File: src/Lib.hs [hunk 1]
```diff
@@ -352,13 +406,17 @@ killZeroes :: [Char] -> [Char]
...
+--- Git Operations ---
+----------------------
+
 data GithubConfig = GC {
       username :: Text
     , oauth    :: Text
     } deriving (Show, Eq, Ord)
 
-githubSpec :: ValueSpec GithubConfig
+githubSpec :: ValueSpecs GithubConfig
```
Comment (shapr) [inline src/Lib.hs:419]: This is specific to a pre-1.0 version of config-schema, but it's easy to fix.

**Review by shapr — APPROVED** [review]
yup!

**Comment by shapr** [comment]
This fixes #16

**Pull request merged**